--- xkeyboard-config-1.1/symbols/ee~	2007-07-16 01:12:12.000000000 +0300
+++ xkeyboard-config-1.1/symbols/ee	2007-12-26 13:10:23.000000000 +0200
@@ -21,7 +21,6 @@
     key <AE11>	{ [      plus,   question,    backslash, questiondown ]	};
     key <AE12>	{ [dead_acute, dead_grave,        grave,   apostrophe ]	};
 
-    key <AD03>	{ [         e,          E,     EuroSign,     EuroSign ]	};
     key <AD11>	{ [udiaeresis, Udiaeresis, dead_diaeresis, dead_abovering ] };
     key <AD12>	{ [    otilde,     Otilde,      section,  dead_macron ]	};
 
@@ -36,6 +35,7 @@
     key <AB09>	{ [    period,      colon,      greater,     division ]	};
 
     include "level3(ralt_switch)"
+    include "eurosign(5)"
 };
 
 partial alphanumeric_keys
@@ -64,7 +64,6 @@
    
     key <AE03>	{ [         3, numbersign,   sterling ]	};
     key <AE04>	{ [         4,     dollar,    section,   currency ]	};
-    key <AE05>	{ [         5,    percent,   EuroSign ]	};
     key <AE08>	{ [         8,   asterisk,    onehalf ]	};
 
     key <AC01>	{ [         a,          A, adiaeresis, Adiaeresis ]	};
@@ -76,4 +75,29 @@
 
     include "level3(ralt_switch)"
     include "kpdl(comma)"
+    include "eurosign(e)"
+};
+
+// Estonian few characters on basic US keymap.
+// by Elan Ruusamäe <glen@delfi.ee>
+partial alphanumeric_keys
+xkb_symbols "us" {
+    include "us(euro)"
+    name[Group1] = "Estonia - U.S. keyboard with Estonian letters";
+
+    // ž, Ž
+    key <AB01>	{ [           z,  Z,          zcaron,     Zcaron     ]	};
+    // ¢
+    key <AB03>	{ [           c,  C,          cent,       cent       ]	};
+    // š, Š
+    key <AC02>	{ [           s,  S,          scaron,     Scaron     ]	};
+    key <AC10>	{ [ semicolon,    colon,      odiaeresis, Odiaeresis ]	};
+    key <AC11>	{ [ quoteright,   quotedbl,   adiaeresis, Adiaeresis ]	};
+    key <AD11>	{ [ bracketleft,  braceleft,  udiaeresis, Udiaeresis ]	};
+    key <AD12>	{ [ bracketright, braceright, otilde,     Otilde     ]	};
+    // £
+    key <AE03>	{ [           3,  numbersign, sterling,   sterling   ]	};
+
+    include "level3(ralt_switch)"
+    include "eurosign(e)"
 };
--- xkeyboard-config-1.1/rules/base.xml.in~	2007-09-26 14:56:42.000000000 +0300
+++ xkeyboard-config-1.1/rules/base.xml.in	2007-12-25 22:45:56.886753769 +0200
@@ -1645,6 +1645,12 @@
             <_description>Dvorak</_description>
           </configItem>
         </variant>
+        <variant>
+          <configItem>
+            <name>us</name>
+            <_description>US keyboard with Estonian letters</_description>
+          </configItem>
+        </variant>
       </variantList>
     </layout>
     <layout>
